icu4c/source/i18n/datefmt.cpp

```diff
@@ -279,9 +279,8 @@ UnicodeString&
 DateFormat::format(UDate date, UnicodeString& appendTo, FieldPosition& fieldPosition) const {
     if (fCalendar != nullptr) {
         UErrorCode ec = U_ZERO_ERROR;
-        const auto* calType = fCalendar->getType();
         // Avoid a heap allocation and corresponding free for the common case
-        if (uprv_strcmp(calType, "gregorian") == 0) {
+        if (GregorianCalendar::getStaticClassID() == fCalendar->getDynamicClassID()) {
```

Comment:
Questions that I had asked on the ticket:
- Why is using getType() bad? It seems like it’s public and advertised.
- Is strcmp() actually slow for short strings that usually differ in the first couple of characters?

Reply:
I think such usage is bad because our code could easily misspell the string